bot/exts/valentines/valentine_zodiac.py

```diff
@@ -19,36 +22,117 @@ class ValentineZodiac(commands.Cog):
 
     def __init__(self, bot: commands.Bot):
         self.bot = bot
-        self.zodiacs = self.load_json()
+        self.zodiacs, self.zodiac_fact = self.load_comp_json()
 
     @staticmethod
-    def load_json() -> dict:
+    def load_comp_json() -> Tuple[dict, dict]:
         """Load zodiac compatibility from static JSON resource."""
-        p = Path("bot/resources/valentines/zodiac_compatibility.json")
-        with p.open(encoding="utf8") as json_data:
-            zodiacs = load(json_data)
-            return zodiacs
+        explanation_file = Path("bot/resources/valentines/zodiac_explanation.json")
+        compatibility_file = Path("bot/resources/valentines/zodiac_compatibility.json")
+        with explanation_file.open(encoding="utf8") as json_data:
+            zodiac_fact = json.load(json_data)
+            for zodiac_data in zodiac_fact.values():
+                zodiac_data['start_at'] = datetime.fromisoformat(zodiac_data['start_at'])
+                zodiac_data['end_at'] = datetime.fromisoformat(zodiac_data['end_at'])
+        with compatibility_file.open(encoding="utf8") as json_data:
+            zodiacs = json.load(json_data)
+        return zodiacs, zodiac_fact
 
-    @commands.command(name="partnerzodiac")
-    async def counter_zodiac(self, ctx: commands.Context, zodiac_sign: str) -> None:
+    def generate_invalidname_embed(self, zodiac: str) -> discord.Embed:
+        """Returns error embed."""
+        embed = discord.Embed()
+        embed.color = Colours.soft_red
+        error_comp = "\n".join(
+            f"`{i}` {zod_name}" for i, zod_name in enumerate(self.zodiac_fact.keys(), start=1)
+        )
+        embed.description = (
+            f"**{zodiac}** is not a valid zodiac sign, here is the list of valid zodiac signs.\n"
+            f"{error_comp}"
+        )
+        log.info("Invalid zodiac name provided.")
+        return embed
+
+    def zodiac_build_embed(self, zodiac: str) -> discord.Embed:
+        """Gives informative zodiac embed."""
```

Comment:
It doesn't really "give" an embed as much as building an embed for either the facts OR error embed.